pesuacademy/pages/faculty.py

```diff
@@ -219,15 +220,56 @@ def get_page(
         designation: Optional[str] = None,
     ) -> list[Professor]:
         urls = self.get_urls_from_campus_and_department(campus, department)
-        # TODO: Scrape the data from the URLs. Use the same session object provided.
-        # TODO: Add search functionality for name: https://staff.pes.edu/atoz/list/?search={name}
-        professors: list[Professor] = list()
+        professors: list[Professor] = []
         for url in urls:
-            faculty_ids = self.get_all_faculty_ids_from_url(session, url, page=1)
-            for faculty_id in faculty_ids:
-                professors.extend(self.get_faculty_by_id(session, faculty_id))
-        if designation is not None:
-            professors = list(
-                filter(lambda x: x.designation == designation, professors)
-            )
```

Comment:
Don't change this part. Later when we add tests, we should be able to individually test a faculty getter function by ID. Move your code changes inside `get_faculty_by_id`.